modules/controllers/queries-controller.ts

```diff
@@ -0,0 +1,109 @@
+/**
+ * Responsible for handling all the queries – can be split based on models
+ */
+import { GqlPoolEvent, GqlPoolJoinExit, GqlPoolSwap, QueryPoolGetEventsArgs } from '../../schema';
+import { prisma } from '../../prisma/prisma-client';
+import { Prisma } from '@prisma/client';
+import { JoinExitEvent, SwapEvent } from '../../prisma/prisma-types';
+
+const parseJoinExit = (event: JoinExitEvent): GqlPoolJoinExit => {
+    return {
+        __typename: 'GqlPoolJoinExit',
+        amounts: event.payload.tokens.map((token: any) => ({
+            address: token.address,
+            amount: token.amount,
+            valueUSD: token.amountUsd,
+        })),
+        ...event,
+        timestamp: event.blockTimestamp,
+        sender: event.userAddress,
+    };
+};
+
+const parseSwap = (event: SwapEvent): GqlPoolSwap => {
+    return {
+        __typename: 'GqlPoolSwap',
+        ...event,
+        sender: event.userAddress,
+        timestamp: event.blockTimestamp,
+        tokenIn: event.payload.tokenIn.address,
+        tokenAmountIn: event.payload.tokenIn.amount,
+        tokenInData: {
+            ...event.payload.tokenIn,
+        },
+        tokenOut: event.payload.tokenOut.address,
+        tokenAmountOut: event.payload.tokenOut.amount,
+        tokenOutData: {
+            ...event.payload.tokenOut,
+        },
+    };
+};
+
+export function QueriesController(tracer?: any) {
```

Comment:
This wont scale, we'll need to split that into multiple queries controllers.Though I am not even sure whether these should be controllers or whether we just leave them in the pool service.

Reply:
yes, agree, i consider this a placeholder, lets give it a revamp once we collect more actions. i like loaders / mutations split by domain, same goes for jobs controller.